### PR TITLE
[TACHYON-624] Distinguish between remote and local writes for client metrics

### PR DESCRIPTION
**Pull request by gpang**
https://tachyon.atlassian.net/browse/TACHYON-624

This PR adds client metrics for remote writes. They essentially mirror the statistics for "local blocks written" and "local bytes written".

### CODE REVIEW

File: common/src/main/java/tachyon/Constants.java [hunk 1]
```diff
@@ -224,15 +224,16 @@
   public static final boolean DEBUG = Boolean.valueOf(System.getProperty(TACHYON_DEBUG, "false"));
 
   public static final long CLIENT_METRICS_VERSION = 1L;
-  public static final int CLIENT_METRICS_SIZE = 9;
+  public static final int CLIENT_METRICS_SIZE = 11;
```
Comment (apc999) [inline common/src/main/java/tachyon/Constants.java]: this is becoming a ".proto" file. We should really convert it to protobuf

Comment (gpang) [inline common/src/main/java/tachyon/Constants.java]: Is someone already converting this to a proto?

**Comment by apc999** [comment]
+1 LGTM. I think we should test out using protobuf to serialize/deserialize metrics values. It worths a advanced JIRA

**Comment by haoyuan** [comment]
@carsonwang

File: clients/unshaded/src/main/java/tachyon/client/LocalBlockOutStream.java [hunk 2]
```diff
@@ -160,6 +161,7 @@ public void close() throws IOException {
       mCloser.close();
       mTachyonFS.cacheBlock(mBlockId);
       mClosed = true;
+      mTachyonFS.getClientMetrics().incBlocksWrittenLocal(1);
```
Comment (carsonwang) [inline clients/unshaded/src/main/java/tachyon/client/LocalBlockOutStream.java]: Do you need check if mWrittenBytes > 0 here? Only caches the block and increments the metrics if written bytes > 0 ?

Comment (gpang) [inline clients/unshaded/src/main/java/tachyon/client/LocalBlockOutStream.java]: Thanks, I added the "mWrittenBytes > 0" check.

File: servers/src/main/java/tachyon/worker/block/BlockDataManager.java [hunk 3]
```diff
@@ -434,10 +434,12 @@ public void userHeartbeat(long userId, List<Long> metrics) {
       mWorkerSource.incBlocksReadLocal(metrics.get(Constants.BLOCKS_READ_LOCAL_INDEX));
       mWorkerSource.incBlocksReadRemote(metrics.get(Constants.BLOCKS_READ_REMOTE_INDEX));
       mWorkerSource.incBlocksWrittenLocal(metrics.get(Constants.BLOCKS_WRITTEN_LOCAL_INDEX));
+      mWorkerSource.incBlocksWrittenRemote(metrics.get(Constants.BLOCKS_WRITTEN_REMOTE_INDEX));
       mWorkerSource.incBytesReadLocal(metrics.get(Constants.BYTES_READ_LOCAL_INDEX));
       mWorkerSource.incBytesReadRemote(metrics.get(Constants.BYTES_READ_REMOTE_INDEX));
       mWorkerSource.incBytesReadUfs(metrics.get(Constants.BYTES_READ_UFS_INDEX));
       mWorkerSource.incBytesWrittenLocal(metrics.get(Constants.BYTES_WRITTEN_LOCAL_INDEX));
+      mWorkerSource.incBytesWrittenRemote(metrics.get(Constants.BYTES_WRITTEN_REMOTE_INDEX));
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/worker/block/BlockDataManager.java]: Would it make sense to switch on the version number of the heartbeat? This way 0.7 clients can communicate with the new server and still contribute (though arguably incorrect) metrics?

**Comment by gpang** [comment]
I will close this PR and reopen it on the 0.7 RC branch.

**Pull request closed without merging**